lib/features/cache_save.js

```diff
@@ -0,0 +1,53 @@
+import Debug from 'debug';
+import fs from 'mz/fs';
+import _ from 'lodash';
+import path from 'path';
+import Promise from 'promise';
+import slugid from 'slugid';
+// import tarStream from 'tar-stream';
+import tar from 'tar-fs';
+import uploadToS3 from '../upload_to_s3';
+import waitForEvent from '../wait_for_event';
+import zlib from 'zlib';
+
+let debug = Debug('docker-worker:features:cache-save');
+
+//must be changed if the delimimter in lib/volume_cache.js is changed
+let KEY_DELIMITER = '::';
+
+export default class DockerSave {
+  async killed (task) {
+    // var packStream = tarStream.pack();
+    await Promise.all(task.volumeCaches.map(async (cacheStr) => {
+      let cache = cacheStr.split(KEY_DELIMITER);
```

Comment:
Cache keys can be split... but this is not the intention... It's a bad idea... we should add to an extra return value to `buildVolumeBindings` and patch `task.js` to attach the list of `cacheInstance` returned by `volumeCache.get(volumeName);`. And the use the `cacheInstance.path` property instead..

At the moment this creates a situation where we can't change how cache keys are constructed... Ideally, they should just have been uuids.